dask/array/backends.py

```diff
@@ -120,17 +123,24 @@ def _tensordot(a, b, axes=2):
 @concatenate_lookup.register_lazy("cupy")
 @nannumel_lookup.register_lazy("cupy")
 @numel_lookup.register_lazy("cupy")
+@to_numpy_dispatch.register_lazy("cupy")
 def register_cupy():
     import cupy
 
-    from dask.array.dispatch import percentile_lookup
-
     concatenate_lookup.register(cupy.ndarray, cupy.concatenate)
     tensordot_lookup.register(cupy.ndarray, cupy.tensordot)
     percentile_lookup.register(cupy.ndarray, percentile)
     numel_lookup.register(cupy.ndarray, _numel_arraylike)
     nannumel_lookup.register(cupy.ndarray, _nannumel)
 
+    @to_numpy_dispatch.register(cupy.ndarray)
+    def cupy_to_numpy(data, **kwargs):
+        return cupy.asnumpy(data, **kwargs)
+
+    @to_cupy_dispatch.register(np.ndarray)
+    def numpy_to_cupy(data, **kwargs):
+        return cupy.asarray(data, **kwargs)
+
```

Comment:
(Non-blocking comment) This could get pretty involved as the number of backends grow. I'm curious:

- Can our `asarray_safe` utility somehow help simplify this?
- Can we drop the need for the dispatch objects and just call `np.asarry(...)` for converting arrays to the numpy backend / `cupy.asarray` for converting to the cupy backend? My guess here is "no" but I just wanted to make sure we considered the possibility here 

In practice, I'm not super concerned about this as I expect the number of backends won't get out of hand

Reply:
>This could get pretty involved as the number of backends grow

Yeah, I agree that the "all-to-all" approach wouldn't scale to a large number of backends very well. However, I don't think the number of backends will grow much.  I also wouldn't expect all backend-to-backend conversions to be equally important.

>Can our asarray_safe utility somehow help simplify this?

I don't *think* so, but I'll be happy to modify the approach used here if so. As far as I understand, we *could* directly use `asarray_safe` in `CupyBackendEntrypoint.to_backend_dispatch` to convert from numpy to cupy. However, cupy to numpy conversion will raise an error (e.g. `TypeError: Implicit conversion to a NumPy array is not allowed`). Also, even if there are shortcuts for the array backend, the more-verbose pattern is probably needed for pandas<->cudf.

>Can we drop the need for the dispatch objects and just call `np.asarry(...)`...

Pretty sure the challenges are exactly the same as `asarray_safe`.

Reply:
Rick is correct according to `asarray_safe`. What we could do instead is a "super" `asarray_safe`, which would catch the exception above and then use `.get()` to still convert to CuPy->NumPy. I'm not entirely sure this make things much simpler, so this is just an idea.